src/headers/tomcrypt_pk.h

```diff
@@ -529,6 +529,10 @@ typedef struct ltc_asn1_list_ {
    unsigned long size;
    /** The used flag, this is used by the CHOICE ASN.1 type to indicate which choice was made */
    int           used;
+   /** Flag used to indicate optional items in ASN.1 sequences */
+   int           optional;
+   /** Flag used to indicate context specific tags on ASN.1 sequence items */
+   unsigned char tag;
```

Comment:
I'd say `unsigned char` is inappropriate here as
1. `0` is a valid context-specific tag
2. tags can be longer than 1 byte (yes this is currently nowhere taken into account so I think it's the time now to do it better)

I think we have two possibilities now:
1. introduce a flags field that would then for now include _Class_ and _P/C_ of the ASN.1 identifier and _optional_ of our implementation. Then have an `unsigned long tag;` which is taken into account when _Class_ indicates a context-specific tag.
2. we can use one of the signed types where `-1` indicates `invalid/unused`